jobfunnel/backend/scrapers/monster.py

```diff
@@ -260,15 +260,32 @@ def _get_search_url(self, method: Optional[str] = 'get',
               all previous jobs as we go.
         """
         if method == 'get':
-            return (
-                'https://www.monster.{}/jobs/search/?{}q={}&where={}__2C-{}'
-                    '&rad={}'.format(
+            search_url = 'https://www.monster.{}/jobs/search/?{}q={}'.format(
                         self.config.search_config.domain,
                         f'page={page}&' if page > 1 else '',
                         self.query,
-                        self.config.search_config.city.replace(' ', '-'),
+                )
+            # search countrywide supposing one is open to relocating within
+            # the country (includes remote)
+            if self.config.search_config.city is None and \
+                self.config.search_config.province_or_state is None and \
+                self.config.search_config.remote_within_country is False:
+                return search_url
+            # search remote within the country
+            elif self.config.search_config.remote_within_country is True:
+                return search_url + '-remote'
+            # search by state or province, does not factor in city or radius
+            elif self.config.search_config.city is None and \
+                self.config.search_config.province_or_state and \
+                self.config.search_config.remote_within_country is False:
+                return search_url + '&where={}'.format(
                         self.config.search_config.province_or_state,
-                        self._convert_radius(self.config.search_config.radius)
+                )
+            return (
+                search_url + '&where={}__2C-{}&rad={}'.format(
+                    self.config.search_config.city.replace(' ', '-'),
+                    self.config.search_config.province_or_state,
+                    self._convert_radius(self.config.search_config.radius)
```

Comment:
I think this might overcomplicate things at this stage. The config builder should process all user provided input from the CLI and YAML and create a valid configuration. This configuration should then be directly processed by the scrapers.

The potential issue I see with complicating the logic for setting up the URL is that it is a lot harder to extend the scrapers with different LOCALES. Right now, we try to optimise the scraper classes such that it is easy to support more LOCALES.